tests/system/libraries/nvdaRobotLib.py

```diff
@@ -0,0 +1,219 @@
+from systemTestUtils import _blockUntilConditionMet
+
+builtIn = BuiltIn()  # type: BuiltIn
+process = builtIn.get_library_instance('Process')  # type: Process
+opSys = builtIn.get_library_instance('OperatingSystem')  # type: OperatingSystem
+
+spyServerPort = 8270  # is `registered by IANA` for remote server usage. Two ASCII values:'RF'
+spyServerURI = 'http://127.0.0.1:{}'.format(spyServerPort)
+spyAlias = "nvdaSpy"
+
+# robot is expected to be run from the NVDA repo root directory. We want all repo specific
+# paths to be relative to this. This would allow us to change where it is run from if we decided to.
+repoRoot = _abspath("./")
+whichNVDA = builtIn.get_variable_value("${whichNVDA}", "source")
+if whichNVDA == "source":
+	NVDACommandPathToCheckExists = _pJoin(repoRoot, "source/nvda.pyw")
+	baseNVDACommandline = "pythonw "+NVDACommandPathToCheckExists
+elif whichNVDA == "installed":
+	NVDACommandPathToCheckExists = _pJoin(_expandvars('%PROGRAMFILES%'),'nvda','nvda.exe')
+	baseNVDACommandline='"%s"' % NVDACommandPathToCheckExists
+else:
+	raise AssertionError("robot should be given argument `-v whichNVDA [source|installed]")
+
+# Paths
+systemTestSourceDir = _pJoin(repoRoot, "tests", "system")
+tempDir = tempfile.gettempdir()
+opSys.directory_should_exist(tempDir)
+nvdaProfileWorkingDir = _pJoin(tempDir, "nvdaProfile")
+nvdaLogFilePath = _pJoin(nvdaProfileWorkingDir, 'nvda.log')
+systemTestSpyAddonName = "systemTestSpy"
+testSpyPackageDest = _pJoin(nvdaProfileWorkingDir, "globalPlugins")
+outDir = builtIn.get_variable_value("${OUTPUT DIR}")
+testOutputNvdaLogsDir = _pJoin(outDir, "nvdaTestRunLogs")
+
+
+def _findDepPath(depFileName, searchPaths):
+	import os
+	for path in searchPaths:
+		filePath = _pJoin(path, depFileName+".py")
+		if os.path.isfile(filePath):
+			return filePath
+		elif os.path.isfile(_pJoin(path, depFileName, "__init__.py")):
+			return _pJoin(path, depFileName)
+	raise AssertionError("Unable to find required system test spy dependency: {}".format(depFileName))
+
+# relative to the python path
+requiredPythonImportsForSystemTestSpyPackage = [
+	r"robotremoteserver",
+	r"SimpleXMLRPCServer",
+	r"xmlrpclib",
+]
+
+def _createNvdaSpyPackage():
+	import os
+	searchPaths = sys.path
+	profileSysTestSpyPackageStagingDir = _pJoin(tempDir, systemTestSpyAddonName)
+	# copy in required dependencies, the addon will modify the python path
+	# to point to this sub dir
+	spyPackageLibsDir = _pJoin(profileSysTestSpyPackageStagingDir, "libs")
+	opSys.create_directory(spyPackageLibsDir)
+	for lib in requiredPythonImportsForSystemTestSpyPackage:
+		libSource = _findDepPath(lib, searchPaths)
+		if os.path.isdir(libSource):
+			opSys.copy_directory(libSource, spyPackageLibsDir)
+		elif os.path.isfile(libSource):
+			opSys.copy_file(libSource, spyPackageLibsDir)
+
+	opSys.copy_file(
+		_pJoin(systemTestSourceDir, "libraries", systemTestSpyAddonName+".py"),
+		_pJoin(profileSysTestSpyPackageStagingDir, "__init__.py")
+	)
+	opSys.copy_file(

```

Comment:
Within the system tests I have preferred to use methods available in the system test framework, since they tend to provide more robustness, checks, and reporting when something is not right.  This makes it easier to track down the problem in the system test log.  Even when things go correctly, there are entries (including timestamps) added to the system test log which can be quite helpful.